source/socket.cc

```diff
@@ -160,6 +136,40 @@ struct addrinfo_container
 	~addrinfo_container() { if (ptr) freeaddrinfo(ptr); };
 };
 
+inline int poll( SocketChannel *chann, int &timeout, std::atomic<bool> &interrupted )
+{
+	if (timeout < 0) timeout = 0;
+	chann->poll.revents = 0;
+#ifdef WB_WINDOWS
+	auto start = tick();
+	int result = WSAPoll(&chann->poll, 1, timeout);
+	timeout -= (int) (tick() - start);
+#else
+	int result;
+	do
+	{
+		auto start = tick();
+		result = ::poll(&chann->poll, 1, 100);
```

Comment:
Use **WSAPoll** and **poll** in this loop to keep the same logic in both platforms.